### PR TITLE
Internal: Create tests for the core PLG [ED-16268]

### PR DESCRIPTION
**Pull request by KingYes**

### CODE REVIEW

File: tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php [hunk 1]
```diff
@@ -529,4 +529,39 @@ private function get_connected_user() {
 
 		return $user;
 	}
+
+	public function test_ge111t_remote_authorize_url() {
```
Comment (louiswol94) [inline tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php:533]:
```suggestion
	public function test_get_remote_authorize_url() {
```
What is the `111`?

File: tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php [hunk 2]
```diff
@@ -529,4 +529,39 @@ private function get_connected_user() {
...
+		$_GET['utm_campaign'] = 'test-campaign';
+		$_GET['utm_not_allowed_param'] = 'test-test';
+
+		$utm_campaign = [
+			'source' => 'transient-source',
+			'medium' => 'transient-medium',
+			'campaign' => 'transient-campaign',
+		];
+
+		set_transient( 'elementor_core_campaign', $utm_campaign );
```
Comment (louiswol94) [inline tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php:546]: Maybe add a const for `elementor_core_campaign`?

File: tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php (hunk 1 above)
Comment (louiswol94) [inline tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php:533]: Is there not already a test for this?

https://github.com/elementor/elementor/blob/2c4254606e62d9ec6ffa644f947d4c0a49ffe4b0/tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php#L479C18-L503

File: tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php [hunk 3]
```diff
@@ -529,4 +529,39 @@ private function get_connected_user() {
...
+
+		// Act
+		$url = $this->app_stub->proxy_get_remote_authorize_url();
+
+		// Assert
+		$parsed_url = parse_url( $url );
+		$parsed_query_params = [];
+		parse_str( $parsed_url['query'], $parsed_query_params );
```
Comment (louiswol94) [inline tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php:554]: Maybe extract to a method?

File: tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php (hunk 2 above)
Comment (KingYes) [reply tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php:546]: No needed. It's based on the 3rd party.

**Review by louiswol94 — APPROVED** [review]

File: tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php (hunk 3 above)
Comment (KingYes) [reply tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php:554]: What exactly do you mean? Two variables have been created.

**Review by asafdl — APPROVED** [review]

File: tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php (hunk 1 above)
Comment (rami-elementor) [reply tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php]: @KingYes

**Pull request merged**